pkg/sqs/sqs.go

```diff
@@ -98,36 +98,37 @@ func (mh *MessageHandler) handleMessage(msg *sqs.ReceiveMessageOutput) {
 	for m := range msg.Messages {
 		log.Debugf("Message body: %s", string(*msg.Messages[m].Body))
 
-		var message util.Message
-		err := json.Unmarshal([]byte(*msg.Messages[m].Body), &message)
-
 		messageHandle := msg.Messages[m].ReceiptHandle
 
-		if err != nil {
-			log.Error(err)
-			mh.deleteConsumedMessage(messageHandle)
-			return
-		}
+		var messageASG util.ASGMessage
+		var messageSpot util.SpotMessage
+		json.Unmarshal([]byte(*msg.Messages[m].Body), &messageASG)
+		json.Unmarshal([]byte(*msg.Messages[m].Body), &messageSpot)
 
-		if message.EC2InstanceId == nil {
+		if messageASG.AutoScalingGroupName != nil && messageASG.EC2InstanceId != nil {
```

Comment:
This double unmarshaling feels odd to me. Unfortunately the only clean way that comes into my mind is to split the queues.

Reply:
we pay per message (not per queue) so that would work. i also feel like this is an ugly solution but afaik there is no nicer way of solving this in golang. having two queues feels a bit overpowered to me as well.